lib/jsonapi/active_model_error_serializer.rb

```diff
@@ -12,35 +12,15 @@ class ActiveModelErrorSerializer < ErrorSerializer
     end
 
     attribute :code do |object|
-      _, error_hash = object
-      code = error_hash[:error] unless error_hash[:error].is_a?(Hash)
-      code ||= error_hash[:message] || :invalid
-      # `parameterize` separator arguments are different on Rails 4 vs 5...
-      code.to_s.delete("''").parameterize.tr('-', '_')
+      object.type.to_s.delete("''").parameterize.tr('-', '_')
     end
 
-    attribute :detail do |object, params|
-      error_key, error_hash = object
-      errors_object = params[:model].errors
-
-      # Rails 4 provides just the message.
-      if error_hash[:error].present? && error_hash[:error].is_a?(Hash)
-        message = errors_object.generate_message(
-          error_key, nil, error_hash[:error]
-        )
-      elsif error_hash[:error].present?
-        message = errors_object.generate_message(
-          error_key, error_hash[:error], error_hash
-        )
-      else
-        message = error_hash[:message]
-      end
-
-      errors_object.full_message(error_key, message)
+    attribute :detail do |object, _params|
+      object.full_message
     end
 
     attribute :source do |object, params|
-      error_key, _ = object
+      error_key = object.attribute
```

Comment:
it seems before this change an array was supported here as `object` but now it is not.  Is there a reason for the change in behavior there?

Reply:
No, that was certainly not intended, but there was also no test for it. If you depend on that behaviour, as a spec and an `Array.wrap` or similar as a PR.

Reply:
if this is more correct i can update on my end.  we were building up these errors manually (for some unknown reason) which this suddenly broke.